src/application/service/CreateAccessPass.ts

```diff
@@ -0,0 +1,38 @@
+import { UserId } from '../../domain/model/user/UserId';
+import { AccessPass } from '../../domain/model/accessPass/AccessPass';
+import { AccessPassRepository } from '../../domain/model/accessPass/AccessPassRepository';
+import { SharingCodeRepository } from '../../domain/model/sharingCode/SharingCodeRepository';
+
+export class CreateAccessPass {
+  constructor(
+    private accessPassRepository: AccessPassRepository,
+    private sharingCodeRepository: SharingCodeRepository
+  ) {}
+
+  public async withSharingCode(
+    code: string,
+    userId: string
+  ): Promise<AccessPass> {
+    const sharingCode = await sharingCodeRepository.findByCode(code);
```

Comment:
here we should also check if the sharing code is not found?

Reply:
Yes, since done in my latest 👍